libvast/src/system/index.cpp

```diff
@@ -345,76 +345,86 @@ caf::error index_state::load_from_disk() {
-                                       static_cast<store_actor>(global_store),
-                                       filesystem, path);
-          self->request(partition, caf::infinite, atom::erase_v)
-            .then(
-              [uuid](atom::done) {
-                VAST_DEBUG("index erased partition {} during startup", uuid);
-              },
-              [uuid](const caf::error& e) {
-                VAST_WARN("index failed to erase partition {} during startup: "
-                          "{}",
-                          uuid, e);
-              });
+                               fmt::format("failed to list directory contents "
+                                           "of {}: {}",
+                                           dir, err.message()));
+      for (auto const& entry : transforms_dir_iter) {
+        if (entry.path().extension() != ".marker")
+          continue;
+        auto chunk = vast::chunk::mmap(entry.path());
+        if (!chunk) {
+          VAST_WARN("{} failed to mmap chunk at {}: {}", *self, entry.path(),
+                    chunk.error());
+          continue;
+        }
+        auto maybe_flatbuffer
+          = vast::flatbuffer<vast::fbs::PartitionTransform>::make(
+            std::move(*chunk));
+        if (!maybe_flatbuffer) {
+          VAST_WARN("{} failed to open transform {}: {}", *self, entry.path(),
+                    err.message());
+          continue;
+        }
+        auto& transform_flatbuffer = *maybe_flatbuffer;
+        if (transform_flatbuffer->transform_type()
+            != vast::fbs::partition_transform::PartitionTransform::v0) {
+          VAST_WARN("{} detected unknown transform version at {}", *self,
+                    entry.path());
+          continue;
+        }
+        auto const* transform_v0 = transform_flatbuffer->transform_as_v0();
+        for (auto const* id : *transform_v0->input_partitions()) {
+          auto uuid = vast::uuid::from_flatbuffer(*id);
+          auto path = partition_path(uuid);
+          if (std::filesystem::exists(path), err) {
+            // TODO: In combination with inhomogeneous partitions, this may
+            // result in incorrect index statistics. This depends on whether the
+            // statistics where already updated on-disk before VAST crashed or
+            // not, which is hard to figure out here.
+            auto partition = self->spawn(passive_partition, uuid, accountant,
+                                         static_cast<store_actor>(global_store),
+                                         filesystem, path);
+            self->request(partition, caf::infinite, atom::erase_v)
+              .then(
+                [uuid](atom::done) {
+                  VAST_DEBUG("index erased partition {} during startup", uuid);
+                },
+                [uuid](const caf::error& e) {
+                  VAST_WARN("index failed to erase partition {} during "
+                            "startup: "
```

Comment:
```suggestion
                  VAST_WARN("{} failed to erase partition {} during "
                            "startup: ", *self,
```

For consistency with the other log messages.